pkg/datagatherer/aks/aks.go

```diff
@@ -19,6 +19,13 @@ type AKSDataGatherer struct {
 	credentials   *AzureCredentials
 }
 
+type AKSDataGathererConfig struct {
+	Cluster       string `mapstructure:"cluster"`
+	ResourceGroup string `mapstructure:"resource-group"`
+	Credentials   string `mapstructure:"credentials"`
+	DataPath      string `mapstructure:"data-path"`
```

Comment:
Is this `DataPath` parameter necessary?

I think this was related to loading the data from a file instead of actually instantiating a data-gatherer.

Reply:
The `DataPath` can be specified with all data gatherers to make Preflight use a `local` data gatherer in place of the specific data gatherer (e.g. `gke`, `eks`, etc.) to give 'fake' data. As this struct is used to unmarshal the config it needs to be included. The alternative would be to use a different struct for unmarshalling.

Reply:
>  As this struct is used to unmarshal the config it needs to be included.

I think I am missing something. Where are doing that?

Reply:
wait, I was confusing the datagatherers config with the outputs config. Let me have another look.

Reply:
I see now why you put `DataPath` here.

I think it shouldn't be in the specific datagatherers config but in a general type of config.

As an example, you can create a generic type that embeds all the configuration types and adds the `DataPath`.

```
type DataGathererConfig struct {
  *gke.Config
  *eks.Config
  *aks.Config
  DataPath string
}
```

I think something like that might be needed for the changes we want to make for the cert-manager package and allowing several instances of the same data-gatherer, so I am ok to keep this as you wrote it to avoid adding more complexity to this PR.

Reply:
I wonder if it might be cleaner if the config for a DataGatherer to never know about the DataPath?

Might we just run any datagatherer with a datapath as a local data gatherer? e.g. https://github.com/jetstack/preflight/pull/95/files#diff-d1aa418498ff92dc3def120d994c0379R45-R48